staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -149,7 +163,20 @@ type SharedInformer interface {
 	// between any two resyncs may be longer than the nominal period
 	// because the implementation takes time to do work and there may
 	// be competing load and scheduling noise.
-	AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration)
+	// It returns a handle for the handler that can be used to remove
+	// the handler again and an error if the handler cannot be added.
+	AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) (*ResourceEventHandlerHandle, error)
+	// RemoveEventHandlerByHandle removes a formerly added event handler given by
+	// its registration handle.
+	// If, for some reason, the same handler has been added multiple
+	// times, only the registration for the given registration handle
+	// will be removed.
+	// It returns an error if the handler cannot be removed,
+	// because the informer is already stopped.
+	// Calling Remove on an already removed handle returns no error
+	// because the handler is finally (still) removed after calling this
+	// method.
+	RemoveEventHandlerByHandle(handle *ResourceEventHandlerHandle) error
```

Comment:
ResourceEventHandlerHandle contains a pointer, so you don't need to take it as a pointer, no?

Reply:
You are right, a value object would work and is even comparable, But returning a value object would require to return a handle even if the operation was failing, because it is not possible to return a `nil`.
This would require to figure out whether it is a valid handle or not.

For a user of the method, typically the concrete type is irrelevant, the user just uses the return value, regardless whether it is a pointer or not. So, for the sake of having a general `nil` I would prefer a pointer.

Reply:
This isn't super important to me, but I don't see why it's important to be able to return a nil. Users of this library shouldn't act on any handle if it is gotten with an error, and I don't think they should need to compare to nil. Also since this has an internal pointer, the zero value should act more or less like nil.

Reply:
The listener stored in the registration handle is used to unregister the handle again. Afterwards it is reset to nil (synchronized under the informer lock). If a value object would be used for the handle, it would not possible anymore to reset this fielld and the handler would always be valid for unregistration. This was the reason for the `IsActive` method I removed on your request. Nevertheless the listener field is still used for this purpose.